dao/story.dao.server.js

```diff
@@ -20,6 +20,25 @@ deleteStory = (storyId) => storyModel.deleteOne({story_id: storyId});
 
 updateStory = (storyID, story) => storyModel.update({story_id: storyID}, {$set: story});
 
+likeStory = (story, userID) => {
+    if(!story.liked_by_users.includes(userID)){
+        story.liked_by_users.push(userID);
+        return story;
+    } else {
+        return story;
+    }
+}
+
+unlikeStory = (story, userID) => {
+    for(var i=0; i<story.liked_by_users.length; i++){
+        if(story.liked_by_users[i] === userID){
+            story.liked_by_users.splice(i, 1);
+            return story;
+        }
+    }
+    return null;
+}
+
```

Comment:
Is it possible to store these as hashmaps instead of an array ? Then we don't have to linearly search for it while removing or adding.
Can that be done in our schema ?

Reply:
As discussed with you, I feel like we should continue using array for now. Later on if the user base increases to a large count, we can think of using a hashmap or some other way to improve search efficiency.